app/views/grades/_exam_curve_form.html.erb

```diff
@@ -1,8 +1,8 @@
 <% questions = @assignment.questions %>
-<% flat_questions = @assignment.flattened_questions %>
+<% non_ec_questions = questions.reject{|q| q["extra"]} %>
+<% flat_questions = @assignment.flattened_questions(non_ec_questions) %>
 <% col_count = flat_questions.count %>
-<% any_parts = (questions.count != col_count) %>
-<% total_score = questions.reject{|q| q["extra"]}.map{|q| q["weight"]}.sum %>
+<% total_score = non_ec_questions.map{|q| q["weight"]}.sum %>
```

Comment:
This line is wrong -- pretty sure this now needs to be
```
total_score = flat_questions.reject{|q| q["extra"]}.map{|q| q["weight"]}.sum
```
because individual parts of questions could be extra credit, not just entire questions.

Reply:
So the `non_ec_questions` variable is just irrelevant; we should just be asking for `@assignment.flattened_questions.reject {|q| q["extra"]}.map{|q| q["weight"]}`.  Looking at the schema for exam questions: _either_ a question has no parts and can be e.c. or not, _or_ it has parts that can independently be e.c..  So there's no point in filtering the non-e.c. questions first.

Instead, the line in `exam.rb` above should become `flattened_questions(questions).reject{|q| q["extra"]}.map{|q| q["weight"]}`, so that we're using the exact same logic in both places.